### DISCUSSION TITLE
sqlitejdbc.dll without digital signature can conflict with Windows Smart App Control

### DISCUSSION THREAD

**Discussion by ethoswon** [xerial/sqlite-jdbc#1034] [Q&A]
Windows Smart App Control can block an application, if it uses unsigned DLL.
https://support.microsoft.com/en-au/topic/what-is-smart-app-control-285ea03d-fa88-4d56-882e-6698afdb7003

This jdbc driver extracts exports a unsigned DLL during runtime.
xerial/sqlite-jdbc/src/main/resources/org/sqlite/native/Windows/x86_64/sqlitejdbc.dll
xerial/sqlite-jdbc/src/main/resources/org/sqlite/native/Windows/x86/sqlitejdbc.dll

https://github.com/xerial/sqlite-jdbc#how-does-sqlitejdbc-work

So, apps using this driver can be blocked by Windows Smart App Control.

**Comment by gotson** [comment]
You can and should sign the DLLs with your own certificate. Application packagers like [Conveyor](https://conveyor.hydraulic.dev/) can perform in-jar signing of DLLs for such cases.

The project does not provide signed DLLs, for 2 reasons:
1. It is very expensive to buy a signing certificate
2. OSes with controls as you mention expects all the binaries to be signed with the same certificate (yours). Using DLLs signed with another certificate would not solve the problem (i don't know for Windows, but this is the case on macOS).

**Comment by ethoswon** [comment]
Is the DLL written by this project team or 3rd-party?
When we sign it with our own certificate, Isn't there any license issue?

**Reply by gotson** [reply]
the dll is generated by this project, yes. The license of the project is documented.
